File: webapp-net/SmartTarget/Models/SmartTargetItem.cs
using Sdl.Web.Common.Configuration;
using Sdl.Web.Common.Models;
using Sdl.Web.Tridion.ContentManager;

namespace Sdl.Web.Modules.SmartTarget.Models
{
    public class SmartTargetItem
    {
        private readonly Localization _localization;

        public string RegionName { get; set; }

        public string PromotionId { get; set; }

        public string ComponentUri { get; set; }

        public string TemplateUri { get; set; }
        
        public EntityModel Entity 
        {
            get
            {
                TcmUri componentUri = new TcmUri(ComponentUri);
                TcmUri templateUri = new TcmUri(TemplateUri);
                return SiteConfiguration.ContentProvider.GetEntityModel(string.Format("{0}-{1}", componentUri.ItemId, templateUri.ItemId), _localization);
            }
        }

        public SmartTargetItem(Localization localization)
        {
            _localization = localization;
        }
    }
}

File: webapp-java/dxa-module-smarttarget/src/main/java/com/sdl/dxa/modules/smarttarget/model/entity/smarttarget/SmartTargetItem.java
package com.sdl.dxa.modules.smarttarget.model.entity.smarttarget;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.sdl.webapp.common.api.content.ContentProvider;
import com.sdl.webapp.common.api.content.ContentProviderException;
import com.sdl.webapp.common.api.localization.Localization;
import com.sdl.webapp.common.api.model.EntityModel;
import com.sdl.webapp.common.exceptions.DxaException;
import com.sdl.webapp.common.util.ApplicationContextHolder;
import lombok.AccessLevel;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import lombok.extern.slf4j.Slf4j;

/**
 * <p>SmartTargetItem class.</p>
 */
@EqualsAndHashCode
@ToString
@Slf4j
public class SmartTargetItem {

    private final Localization localization;

    @JsonProperty("Entity")
    private EntityModel entity;

    @Setter(AccessLevel.PRIVATE)
    @Getter
    @JsonProperty("EntityId")
    private String entityId;

    /**
     * <p>Constructor for SmartTargetItem.</p>
     *
     * @param entityId     a {@link java.lang.String} object.
     * @param localization a {@link com.sdl.webapp.common.api.localization.Localization} object.
     */
    public SmartTargetItem(String entityId, Localization localization) {
        this.localization = localization;
        this.entityId = entityId;
    }

    /**
     * <p>Getter for the field <code>entity</code>.</p>
     *
     * @return a {@link com.sdl.webapp.common.api.model.EntityModel} object.
     */
    public EntityModel getEntity() {
        if (this.entity == null) {
            ContentProvider contentProvider = ApplicationContextHolder.getContext().getBean(ContentProvider.class);
            try {
                this.entity = contentProvider.getEntityModel(entityId, localization);
            } catch (ContentProviderException | DxaException e) {
                log.warn("EntityModel not found for entity id {}", entityId, e);
            }
        }
        return entity;
    }
}
